src/nvim/fileio.c

```diff
@@ -6792,21 +6804,24 @@ char_u *getnextac(int c, void *cookie, int indent)
   return retval;
 }
 
-/*
- * Return TRUE if there is a matching autocommand for "fname".
- * To account for buffer-local autocommands, function needs to know
- * in which buffer the file will be opened.
- */
-int has_autocmd(event_T event, char_u *sfname, buf_T *buf)
+/// Return true if there is a matching autocommand for "fname".
+/// To account for buffer-local autocommands, function needs to know
+/// in which buffer the file will be opened.
+///
+/// @param event  The event that occured.
+/// @param sfname The filename the event occured in.
+/// @param buf    The buffer the file is open in
+bool has_autocmd(event_T event, char_u *sfname, buf_T *buf)
```

Comment:
This one can be also FUNC_ATTR_PURE, but only because it may be subject to the same optimizations (actually one cannot consider this one pure). I think it is safer to write here just FUNC_ATTR_WARN_UNUSED_RESULT and nothing else.